Sources/Tools/SoundDecompress/plOldAudioFileReader.cpp

```diff
@@ -0,0 +1,156 @@
+JPEG Group JPEG library, Microsoft Windows Media SDK, or Apple QuickTime SDK
+(or a modified version of those libraries),
+containing parts covered by the terms of the Bink SDK EULA, 3ds Max EULA,
+PhysX SDK EULA, DirectX SDK EULA, OpenSSL and SSLeay licenses, IJG
+JPEG Library README, Windows Media SDK EULA, or QuickTime SDK EULA, the
+licensors of this Program grant you additional
+permission to convey the resulting work. Corresponding Source for a
+non-source form of such a combination shall include the source code for
+the parts of OpenSSL and IJG JPEG Library used as well as that of the covered
+work.
+
+You can contact Cyan Worlds, Inc. by email legal@cyan.com
+ or by snail mail at:
+      Cyan Worlds, Inc.
+      14617 N Newport Hwy
+      Mead, WA   99021
+
+*==LICENSE==*/
+//////////////////////////////////////////////////////////////////////////////
+//                                                                          //
+//  plOldAudioFileReader                                                    //
+//                                                                          //
+//// Notes ///////////////////////////////////////////////////////////////////
+//                                                                          //
+//  3.5.2001 - Created by mcn.                                              //
+//                                                                          //
+//////////////////////////////////////////////////////////////////////////////
+
+#include "HeadSpin.h"
+#include "plOldAudioFileReader.h"
+
+#include "plAudioCore/plBufferedFileReader.h"
+#include "plAudioCore/plCachedFileReader.h"
+#include "plAudioCore/plFastWavReader.h"
+#include "plAudioCore/plOGGCodec.h"
+#include "plAudioCore/plWavFile.h"
+
+#define kCacheDirName   "streamingCache"
+
+plAudioFileReader* plOldAudioFileReader::CreateReader(const plFileName& path, plAudioCore::ChannelSelect whichChan, StreamType type)
+{
+    ST::string ext = path.GetFileExt();
+
+    if (type == kStreamWAV) {
+        bool isWav = (ext.compare_i("wav") == 0);
+        // We want to stream a wav off disk, but this is a compressed file.
+        // Get the uncompressed path. Ignore the requested channel, since it 
+        // will have already been split into two files if that is necessary.
+        if (!isWav) {
+            plFileName cachedPath = IGetCachedPath(path, whichChan);
+            std::unique_ptr<plAudioFileReader> r = std::make_unique<plFastWAV>(cachedPath, plAudioCore::kAll);
+            if (!r->IsValid()) {
+                // So we tried to play a cached file and it didn't exist
+                // Oops... we should cache it now
+                ICacheFile(path, true, whichChan);
+                return new plFastWAV(cachedPath, plAudioCore::kAll);
+            }
+            return r.release();
+        }
+
+        return new plFastWAV(path, whichChan);
+    }
+    else if (type == kStreamRAM)
```

Comment:
if braces are used in one branch of an `if... else`, they should be used by all branches.